zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/DocumentenApiClient.kt

```diff
@@ -72,6 +75,49 @@ class DocumentenApiClient(
         return result
     }
 
+     fun storeDocumentInParts(
+        authentication: DocumentenApiAuthentication,
+        baseUrl: URI,
+        request: BestandsdelenRequest,
+        createDocumentResult: CreateDocumentResult,
+        bestandsnaam: String
+    ) {
+         // Inside the CreateDocumentResult there is an array of bestandsdelen.
+         // Each bestandsdeel needs to be sent separately
+         // So the documenten api determines the amount (and size) of chunks, not this application.
+         logger.info { "Starting upload of file $bestandsnaam in ${createDocumentResult.bestandsdelen.size} chunks" }
+
+         createDocumentResult.bestandsdelen.forEach { bestandsdeel ->
+             logger.debug { "Sending chunk #${bestandsdeel.volgnummer} for a size of ${bestandsdeel.omvang} bytes" }
+
+             val chunk = ByteArray(bestandsdeel.omvang)
```

Comment:
This complete piece of code can move intp FileUploadPart if you ask me.

`
val chunk = ByteArray(bestandsdeel.omvang)
val bytesRead = request.inhoud.read(chunk)

check (bytesRead == chunk.size) {
                 "Failed to read all the bytes to upload. The expected omvang is larger than the bytes available."
}
`